src/main/scala/junctions/asyncqueue.scala

```diff
@@ -33,15 +39,19 @@ class AsyncQueueSource[T <: Data](gen: T, depth: Int, sync: Int, clockIn: Clock,
     val mem  = Vec(depth, gen).asOutput
   }
 
-  val mem = Reg(Vec(depth, gen))
+  val mem = Reg(Vec(depth, gen)) //This does NOT need to be asynchronously reset.
   val widx = GrayCounter(bits+1, io.enq.fire())
   val ridx = AsyncGrayCounter(io.ridx, sync)
   val ready = widx =/= (ridx ^ UInt(depth | depth >> 1))
 
   val index = if (depth == 1) UInt(0) else io.widx(bits-1, 0) ^ (io.widx(bits, bits) << (bits-1))
   when (io.enq.fire() && !reset) { mem(index) := io.enq.bits }
-  io.enq.ready := RegNext(ready, Bool(false))
-  io.widx := RegNext(widx, UInt(0))
+  val ready_reg = AsyncResetReg(ready, 0)
+  io.enq.ready := ready_reg
+
+  val widx_reg = AsyncResetReg(widx, 0)
+  io.widx := widx_reg
+
```

Comment:
ah, this was the remnant of my having actual module instantiations before I realized there were nice objects now. I'll tidy up.